src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -84,10 +105,22 @@ internal class PackageRestoreDataSource : ChainedProjectValueDataSourceBase<Rest
             _fileSystem = fileSystem;
             _logger = logger;
             _projectDependentFileChangeNotificationService = projectDependentFileChangeNotificationService;
+            _solutionRestoreService4 = solutionRestoreService4;
         }
 
         protected override IDisposable? LinkExternalInput(ITargetBlock<IProjectVersionedValue<RestoreData>> targetBlock)
         {
+            // Register before this project receives any data flows containing possible nominations.
+            // This is needed because we need to register before any nuget restore or before the solution load.
+#pragma warning disable RS0030 // Do not used banned APIs
+            _ = Task.Run(() =>
+            {
+                _ = _solutionRestoreService4
```

Comment:
to monitor whether the task failed, you need return this inner task, and please remove .ConfiguredAwait(false) here. If you don't use 'await', it should never be called.